compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -1705,6 +1690,115 @@ impl<'block> BrilligBlock<'block> {
         }
     }
 
+    fn initialize_constant_array(
+        &mut self,
+        data: &im::Vector<ValueId>,
+        typ: &Type,
+        dfg: &DataFlowGraph,
+        pointer: MemoryAddress,
+    ) {
+        if data.is_empty() {
+            return;
+        }
+        let item_types: &Vec<_> = match typ {
+            Type::Array(item_types, _) => item_types.borrow(),
+            Type::Slice(item_types) => item_types.borrow(),
+            _ => unreachable!("ICE: Array constant is not of array or slice type {typ:?}"),
+        };
+
+        // Find out if we are repeating the same item over and over
+        let mut data_as_unique_items = HashSet::default();
+        for item_index in (0..data.len()).step_by(item_types.len()) {
+            let item: Vec<_> = (0..item_types.len()).map(|i| data[item_index + i]).collect();
+            data_as_unique_items.insert(item);
+        }
```

Comment:
Since we're only checking if there's a singular item in `data_as_unique_items.len() == 1` below, we shouldn't need to create a HashSet of Vecs at all. We can have an optional value containing the first element and iterate each subsequent element until we find one that doesn't match or reach the end and set a flag to indicate the result.

Reply:
Rewrote it to avoid building a hashset, I just check equality with the first element